packages/nextlove/src/generators/utils/dashify-object-keys.ts

```diff
@@ -0,0 +1,13 @@
+type SnakeToDash<T> = {
+  [K in keyof T as K extends string ? SnakeToDashString<K> : K]: T[K]
+}
+
+type SnakeToDashString<S extends string> = S extends `${infer T}_${infer U}`
+  ? `${T}-${SnakeToDashString<U>}`
+  : S
+
+export function dashifyObjectKeys<T extends object>(obj: T): SnakeToDash<T> {
+  return Object.fromEntries(
+    Object.entries(obj).map(([key, value]) => [key.replace(/_/g, "-"), value])
```

Comment:
Ok to implement this here as it's pretty much a direct find/replace. But fwiw this lib exists and I have been using it for a long time in many projects: https://www.npmjs.com/package/change-case